### ISSUE TITLE
Update new project template to metro-react-native-babel-preset

### ISSUE THREAD

**Issue by janicduplessis** [facebook/react-native#20327]
The babel plugin was recently moved to the metro repo and published under `metro-react-native-babel-preset`. We need to make sure to update all references to the old babel plugin as it will no longer be updated. I think the main thing that needs updating is the new project template `.babelrc` and `package.json`.

Note that it it needs to be set as `module:metro-react-native-babel-preset` in `.babelrc` since it does not respect the standard babel preset naming (maybe we could fix that also).

We also need to communicate this properly in the release notes for existing apps.

**Comment by jamsch** [comment]
So this means that `metro-react-native-babel-preset` is going to be a drop in replacement for `babel-preset-react-native@5.x` / Babel 7 RN projects?

**Comment by janicduplessis** [comment]
@jamsch Yes it is exactly the same, it was just moved to the metro repo and published under acnew name.

**Comment by kelset** [comment]
Yes, I guess this needs to be done before 0.57.0 is released (the first RC will be cut today).

As a side-change to this it would be useful to migrate the `.babelrc` config to the new `babel.config.js` since, quoting [from the docs](https://babeljs.io/docs/en/next/babelrc#merging-behavior):

> If Babel finds other .babelrc files while transpiling files in subfolder, it will merge the configuration together.
Note that it can lead to undefined behaviors, and we recommend to use babel.config.js documentation instead.

**Comment by grabbou** [comment]
Let's check this issue once we release RC, we can test the full workflow and see where the references to old babel plugin are and how that affects the project itself.

**Comment by janicduplessis** [comment]
@grabbou It should still work with the old plugin as long as there are no significant changes to it. But yeah, we should try to update it before the next release just in case.

**Comment by rafaellincoln** [comment]
@kelset @grabbou Has not this been done yet?

**Comment by kelset** [comment]
Yes, via this https://github.com/facebook/react-native/pull/20653 (linked above)
